public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -743,7 +743,11 @@ export class LokiDatasource
    * @returns A Promise that resolves to an array of label names represented as MetricFindValue objects.
    */
   async getTagKeys(options?: DataSourceGetTagKeysOptions<LokiQuery>): Promise<MetricFindValue[]> {
-    const result = await this.languageProvider.fetchLabels({ timeRange: options?.timeRange });
+    let streamSelector;
+    if (options?.filters) {
+      streamSelector = `{${options.filters.map((filter) => `${filter.key}${filter.operator}"${escapeLabelValueInSelector(filter.value, filter.operator)}"`).join(',')}}`;
```

Comment:
I just wonder if we can use, for example, `addLabelToQuery()` to a `{}` query instead of this.